src/k8s/pkg/k8s/setup/serviceargs.go

```diff
@@ -2,19 +2,44 @@ package setup
 
 import (
 	"fmt"
+	"os"
+	"regexp"
+	"strings"
 
+	apiv1 "github.com/canonical/k8s/api/v1"
+	"github.com/canonical/k8s/pkg/snap"
 	"github.com/canonical/k8s/pkg/utils"
 )
 
+const argumentPattern = `--?($ARGUMENT_NAME)[[:space:]]*=[[:space:]]*([^[:space:]\n]+)`
```

Comment:
If we are using regex:

```suggestion
var argumentPatternRe = regexp.MustCompile(`--?($ARGUMENT_NAME)[[:space:]]*=[[:space:]]*([^[:space:]\n]+)`)
```

Reply:
`regexp.MustCompile` is called after the `$ARGUMENT_NAME` is replaced to ensure that the regexp is still valid.

Reply:
I went for map approach so this is obsolete anyway